modules/dnn/include/opencv2/dnn/layer.hpp

```diff
@@ -75,6 +77,7 @@ class CV_EXPORTS LayerFactory
 
 private:
     LayerFactory();
+    static std::map<String, int> registeredTypes;
```

Comment:
It is better to move this into implementation.

OpenCV should not define "private" fields in public header.